ciot-web/src/components/AddressSearchBar/AddressSearchBar.js

```diff
@@ -0,0 +1,118 @@
+import { useState, useEffect } from "react";
+import { v4 } from "uuid";
+import PropTypes from "prop-types";
+import { FormControl, FormLabel } from "react-bootstrap";
+import { getAddressData } from "../../helpers/resourceCalls";
+
+export default function AddressSearchBar({
+  setAddress,
+  currentAddress,
+  getCoords,
+  setLocalityName,
+  handleError,
+}) {
+  const [addresses, setAddresses] = useState([]);
+  const [show, setShow] = useState(false);
+
+  const runSearch = async (event) => {
+    if (event.target.value) {
+      setShow(true);
+    }
+    setAddress(event.target.value);
+    try {
+      const addressData = await getAddressData(event.target.value);
+      if (typeof addressData.data === "string") {
+        return false;
+      }
+      setAddresses(addressData.data.features);
+      setLocalityName(addressData.data.features[0].properties.localityName);
+      return true;
+    } catch (error) {
+      setShow(false);
+      return handleError(
+        "Cannot find address data for this address, please try again."
+      );
+    }
+  };
+
+  const setCoordsForSelectedAddress = (address) => {
+    setAddress(address);
+    getCoords(address);
+    setShow(false);
+  };
+  const selectAddress = (e) => {
+    setAddress(e.target.value);
+    setCoordsForSelectedAddress(e.target.value);
+  };
+
+  const list = () => (
+    <ul className="list-unstyled">
+      {addresses.map((addressData) => (
+        <li key={v4()}>
+          <button
+            style={{
+              background: "transparent",
+              padding: "0",
+              border: "0px solid transparent",
+              color: "#000000",
+              fontWeight: "400",
+              marginTop: "0px",
+              paddingLeft: "6px",
+            }}
+            type="button"
+            onClick={(e) => selectAddress(e)}
+            value={addressData.properties.fullAddress}
+          >
+            {addressData.properties.fullAddress}
+          </button>
+        </li>
+      ))}
+    </ul>
+  );
+
+  return (
+    <div style={{ position: "relative" }}>
+      <FormLabel htmlFor="addressSearch" className="font-weight-bold">
+        Enter Address
+      </FormLabel>
+      <FormControl
+        autoComplete="off"
+        placeholder="eg. 635 Humboldt St, Victoria, BC"
+        className=" bcgov-text-input"
+        id="addressSearch"
+        value={currentAddress || ""}
```

Comment:
The update version in react made the store unstable. It is not updating correctly. For instance this component is getting the value from the state using a selector. However, it is not updating properly. I would suggest to get back to the current version and just get the application running separately. Then, we can update the versions if we have the time and budget.